sound/soc/sof/ipc4-topology.c

```diff
@@ -815,6 +817,84 @@ static void sof_ipc4_widget_free_comp_mixer(struct snd_sof_widget *swidget)
 	swidget->private = NULL;
 }
 
+/*
+ * Add the process modules support. The process modules are defined as snd_soc_dapm_effect
+ * modules. Please check enum sof_ipc_process_type for the details.
+ */
+static int sof_ipc4_widget_setup_comp_process(struct snd_sof_widget *swidget)
+{
+	struct snd_soc_component *scomp = swidget->scomp;
+	struct sof_ipc4_process *process;
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
+	struct snd_sof_control *scontrol;
+	struct sof_ipc4_control_data *control_data;
+	void *cfg;
+	int cfg_size;
+	int ret;
+
+	process = kzalloc(sizeof(*process), GFP_KERNEL);
+	if (!process)
+		return -ENOMEM;
+
+	swidget->private = process;
+
+	ret = sof_ipc4_get_audio_fmt(scomp, swidget, &process->available_fmt, true);
+	if (ret)
+		goto err;
+
+	cfg_size = sizeof(struct sof_ipc4_base_module_cfg) +
+		swidget->output_fmt_cnt * sizeof(struct sof_ipc4_audio_format);
+	/* allocate the cfg for process modules */
```

Comment:
"allocate memory for process module config" perhaps?

Reply:
OK